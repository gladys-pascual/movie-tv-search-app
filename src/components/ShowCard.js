import React from "react";
import Rater from "react-rater";
import "react-rater/lib/react-rater.css";

const ShowCard = ({
  media_type,
  name,
  title,
  poster_path,
  first_air_date,
  release_date,
  vote_average,
}) => {
  const date = new Date(media_type === "tv" ? first_air_date : release_date);
  const year = date.getFullYear();
  const day = date.getDate();
  const month_num = date.getMonth();
  const months = [
    "Jan",
    "Feb",
    "Mar",
    "Apr",
    "May",
    "Jun",
    "Jul",
    "Aug",
    "Sep",
    "Oct",
    "Nov",
    "Dec",
  ];
  let month = months[month_num];

  return (
    <div className="show-card">
      <div className="top-wrapper">
        <div className="img-wrapper">
          <img
            src={`https://image.tmdb.org/t/p/w220_and_h330_face${poster_path}`}
            alt={media_type === "tv" ? name : title}
          />
        </div>
        <div className="text-wrapper">
          <h3>{media_type === "tv" ? name : title}</h3>
          <p className="date"> {`${month} ${day} ${year}`} </p>
        </div>
      </div>

      <div className="rating-wrapper">
        <div className="icon">
          <img
            src={
              media_type === "tv"
                ? "https://img.icons8.com/android/20/000000/retro-tv.png"
                : "https://img.icons8.com/ios-filled/20/000000/movie.png"
            }
            alt={media_type === "tv" ? `tv` : `movie`}
          />
        </div>
        <p className="rating">{vote_average}</p>
        <Rater total={5} rating={vote_average / 2} interactive={false} />
      </div>
    </div>
  );
};

export default ShowCard;
