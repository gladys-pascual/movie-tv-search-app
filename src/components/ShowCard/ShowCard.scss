@import "../../assets/variables.scss";

.show-card {
  margin: 10px;
  border-radius: 12px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: space-between;

  .img-wrapper {
    width: 220px;
    height: 330px;

    img {
      border-radius: 12px;
      width: 100%;
    }
  }
  .text-wrapper {
    width: 220px;
    h3 {
      margin-bottom: 0px;
      padding: 0px 16px;
      font-size: 20px;
    }
    h3,
    .date {
      margin-top: 8px;
      text-align: center;
    }
  }
  .rating-wrapper {
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    padding-bottom: 12px;
    .icon {
      padding-right: 12px;
    }
    .rating {
      margin: 3px;
      padding-right: 4px;
    }
  }
}
