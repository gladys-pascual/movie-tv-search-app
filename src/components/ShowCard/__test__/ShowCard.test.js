import React from "react";
import { render } from "@testing-library/react";
import ShowCard from "../ShowCard";

test("renders a TV show card as expected", () => {
  const props = {
    media_type: "tv",
    first_air_date: "2007-09-24",
    name: "The Big Bang Theory",
    poster_path: "/ooBGRQBdbGzBxAVfExiO8r7kloA.jpg",
    vote_average: 7.6,
  };

  const { container } = render(<ShowCard {...props} />);

  expect(container).toMatchSnapshot();
});

test("renders a movie show card as expected", () => {
  const props = {
    media_type: "movie",
    release_date: "2001-11-16",
    title: "Harry Potter and the Philosopher's Stone",
    poster_path: "/wuMc08IPKEatf9rnMNXvIDxqP4W.jpg",
    vote_average: 7.9,
  };

  const { container } = render(<ShowCard {...props} />);

  expect(container).toMatchSnapshot();
});
