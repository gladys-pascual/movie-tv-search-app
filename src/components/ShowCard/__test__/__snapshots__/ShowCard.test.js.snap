// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a TV show card as expected 1`] = `
<div>
  <div
    class="show-card"
  >
    <div
      class="top-wrapper"
    >
      <div
        class="img-wrapper"
      >
        <img
          alt="The Big Bang Theory"
          src="https://image.tmdb.org/t/p/w220_and_h330_face/ooBGRQBdbGzBxAVfExiO8r7kloA.jpg"
        />
      </div>
      <div
        class="text-wrapper"
      >
        <h3>
          The Big Bang Theory
        </h3>
        <p
          class="date"
        >
           
          Sep 24 2007
           
        </p>
      </div>
    </div>
    <div
      class="rating-wrapper"
    >
      <div
        class="icon"
      >
        <img
          alt="tv"
          src="https://img.icons8.com/android/20/000000/retro-tv.png"
        />
      </div>
      <div
        class="react-rater"
      >
        <div>
          <div
            class="react-rater-star is-disabled is-active"
          >
            ★
          </div>
        </div>
        <div>
          <div
            class="react-rater-star is-disabled is-active"
          >
            ★
          </div>
        </div>
        <div>
          <div
            class="react-rater-star is-disabled is-active"
          >
            ★
          </div>
        </div>
        <div>
          <div
            class="react-rater-star is-disabled is-active-half"
          >
            ★
          </div>
        </div>
        <div>
          <div
            class="react-rater-star is-disabled"
          >
            ★
          </div>
        </div>
      </div>
      <p
        class="rating"
      >
        7.6
      </p>
    </div>
  </div>
</div>
`;

exports[`renders a movie show card as expected 1`] = `
<div>
  <div
    class="show-card"
  >
    <div
      class="top-wrapper"
    >
      <div
        class="img-wrapper"
      >
        <img
          alt="Harry Potter and the Philosopher's Stone"
          src="https://image.tmdb.org/t/p/w220_and_h330_face/wuMc08IPKEatf9rnMNXvIDxqP4W.jpg"
        />
      </div>
      <div
        class="text-wrapper"
      >
        <h3>
          Harry Potter and the Philosopher's Stone
        </h3>
        <p
          class="date"
        >
           
          Nov 16 2001
           
        </p>
      </div>
    </div>
    <div
      class="rating-wrapper"
    >
      <div
        class="icon"
      >
        <img
          alt="movie"
          src="https://img.icons8.com/ios-filled/20/000000/movie.png"
        />
      </div>
      <div
        class="react-rater"
      >
        <div>
          <div
            class="react-rater-star is-disabled is-active"
          >
            ★
          </div>
        </div>
        <div>
          <div
            class="react-rater-star is-disabled is-active"
          >
            ★
          </div>
        </div>
        <div>
          <div
            class="react-rater-star is-disabled is-active"
          >
            ★
          </div>
        </div>
        <div>
          <div
            class="react-rater-star is-disabled is-active-half"
          >
            ★
          </div>
        </div>
        <div>
          <div
            class="react-rater-star is-disabled"
          >
            ★
          </div>
        </div>
      </div>
      <p
        class="rating"
      >
        7.9
      </p>
    </div>
  </div>
</div>
`;
